pkg/cmd/cli/cmd/rsh.go

```diff
@@ -155,8 +155,9 @@ func (o *RshOptions) Validate() error {
 // Run starts a remote shell session on the server
 func (o *RshOptions) Run() error {
 	// Insert the TERM into the command to be run
-	term := fmt.Sprintf("TERM=%s", util.Env("TERM", "xterm"))
-	o.Command = append([]string{"env", term}, o.Command...)
-
+	if len(o.Command) == 1 && o.Command[0] == o.Executable {
+		termsh := fmt.Sprintf("TERM=%q %s", util.Env("TERM", "xterm"), o.Executable)
+		o.Command = append(o.Command, "-c", termsh)
```

Comment:
I think anyone passing a custom executable or custom command is probably on their own.  However, I would probably expect this to work for --shell=/bin/bash.  Ultimately this is a losing game, so it's probably ok to only do this substitution in exactly one case - no command, and o.Executable is the default value.